### PR TITLE
MMT-3483: Implement the Collection Association Form

### PR DESCRIPTION
**Pull request by dmistry1**

### CODE REVIEW

File: static/src/js/components/CollectionAssociation/CollectionAssociation.jsx [hunk 1]
```diff
@@ -0,0 +1,405 @@
...
+            onChange={handleChange}
+            formData={searchFormData}
+            formContext={
+              {
+                focusField,
+                setFocusField
+              }
+            }
+          >
+            <BootstrapSelect type="submit" style={{ marginTop: '-100px' }} onClick={handleCollectionSearch}>
```
Comment (dmistry1) [inline static/src/js/components/CollectionAssociation/CollectionAssociation.jsx:373]: Use our custom Button

File: static/src/js/components/CustomRadioWidget/CustomRadioWidget.jsx [hunk 2]
```diff
@@ -50,6 +50,16 @@ const CustomRadioWidget = ({
     title = uiSchema['ui:title']
   }
 
+  let trueOption = null
+  let falseOption = null
+  let showClear = true
+
+  if (uiSchema['ui:options']) {
```
Comment (dmistry1) [inline static/src/js/components/CustomRadioWidget/CustomRadioWidget.jsx:57]: trueOptionLabel same for false

File: static/src/js/components/CustomSelectWidget/CustomSelectWidget.jsx [hunk 3]
```diff
@@ -158,7 +158,7 @@ const CustomSelectWidget = ({
   const { enumOptions: oneOfEnums } = options || {}
 
   const getExistingValue = () => {
-    if (value) {
+    if (value || value >= 0) {
```
Comment (dmistry1) [inline static/src/js/components/CustomSelectWidget/CustomSelectWidget.jsx:161]: try !==value

File: static/src/js/components/Table/Table.jsx [hunk 4]
```diff
@@ -99,7 +99,7 @@ const Table = ({
 
     content.push(rowData)
   } else if (dataLengthExists === 0) {
-    content.push(<tr key="error-banner" className="text-center"><td colSpan={4}>{noDataError}</td></tr>)
+    content.push(<tr key="error-banner" className="text-center"><td colSpan={headers.length}>{noDataError}</td></tr>)
   } else {
     content.push(<tr key="error-banner" className="text-center"><td colSpan={4}>{error}</td></tr>)
```
Comment (dmistry1) [inline static/src/js/components/Table/Table.jsx:104]: use header.length

File: static/src/js/components/DraftPreview/DraftPreview.jsx [hunk 5]
```diff
@@ -169,7 +178,14 @@ const DraftPreview = () => {
   const handlePublish = () => {
     // Calls publish mutation hook
     setLoading(true)
-    publishMutation(derivedConceptType, nativeId)
+    if (derivedConceptType === 'Variable') {
+      const { _private } = ummMetadata || {}
```
Comment (dmistry1) [inline static/src/js/components/DraftPreview/DraftPreview.jsx:182]: use the state variable here

File: static/src/js/components/DraftPreview/DraftPreview.jsx [hunk 6]
```diff
@@ -243,6 +259,17 @@ const DraftPreview = () => {
       }
     >
       <Container id="metadata-form" className="px-0">
+        {
```
Comment (dmistry1) [inline static/src/js/components/DraftPreview/DraftPreview.jsx:262]: move this to the bottom

File: static/src/js/components/CollectionAssociation/CollectionAssociation.jsx [hunk 7]
```diff
@@ -0,0 +1,405 @@
...
+          <BootstrapSelect
+            className="mb-3 m-2"
+            onClick={handleClear}
+            variant="outline-danger"
+          >
+            Clear Collection Association
+          </BootstrapSelect>
+        </Col>
+
+        <h4 className="mt-5">Currently Selected Collection</h4>
```
Comment (dmistry1) [inline static/src/js/components/CollectionAssociation/CollectionAssociation.jsx:354]: Search for COllection

File: static/src/js/components/CustomSelectWidget/CustomSelectWidget.jsx [hunk 8]
```diff
@@ -158,7 +158,7 @@ const CustomSelectWidget = ({
...
       if (typeof value === 'number') {
```
Comment (cgokey) [inline static/src/js/components/CustomSelectWidget/CustomSelectWidget.jsx:162]: do you need line 161, if so, maybe combine, `if (value && typeof value === 'number') {`

Comment (dmistry1) [reply static/src/js/components/CustomSelectWidget/CustomSelectWidget.jsx:162]: I tried that but no luck. This did not work in the case where the value is a string. 


```
    if (value !== null) {
      if (typeof value === 'number') {
        return oneOfEnums[value]
      }

      return {
        value,
        label: value
      }
    }

    return null
  }
```

the snippet above worked as expected and covered all of the cases

**Review by cgokey — APPROVED** [review]

**Review by tle1989 — APPROVED** [review]

File: static/src/js/components/CollectionAssociationPreviewProgress/CollectionAssociationPreviewProgress.scss [hunk 9]
```diff
@@ -0,0 +1,21 @@
+.collection-association-preview-progress {
```
Comment (trevorlang) [inline static/src/js/components/CollectionAssociationPreviewProgress/CollectionAssociationPreviewProgress.scss:1]: I think some of this CSS looks funky but I don't want to hold this up. Would you mind writing a ticket to circle back and add a comment here?

File: static/src/js/components/DraftPreview/DraftPreview.jsx [hunk 10]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable react/jsx-props-no-spreading */
```
Comment (trevorlang) [inline static/src/js/components/DraftPreview/DraftPreview.jsx:1]: We dont like these whole file disables :)

File: static/src/js/schemas/collectionAssociation.js [hunk 11]
```diff
@@ -0,0 +1,201 @@
...
+    ProviderFilter: {
+      type: 'boolean',
+      // Title: 'Provider Filter',
+      description: 'This is the radio-description',
+      default: true
+    }
+  },
+  required: [
+    'SearchField'
+    // 'ProviderFilter'
```
Comment (trevorlang) [inline static/src/js/schemas/collectionAssociation.js:20]: Remove?

**Review by trevorlang — APPROVED** [review]

**Pull request merged**